src/test/java/org/junitpioneer/jupiter/json/JsonFileSourceArgumentsProviderTests.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2016-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.json;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.junitpioneer.testkit.assertion.PioneerAssert.assertThat;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.platform.commons.PreconditionViolationException;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.support.descriptor.MethodSource;
+import org.junit.platform.testkit.engine.Event;
+import org.junitpioneer.testkit.ExecutionResults;
+import org.junitpioneer.testkit.PioneerTestKit;
+
+/**
+ * Tests for {@link JsonFileArgumentsProvider}
+ */
+class JsonFileSourceArgumentsProviderTests {
+
+	@Test
+	void assertAllValuesSupplied() {
+		ExecutionResults results = PioneerTestKit.executeTestClass(JsonFileSourceTestCases.class);
+
+		Map<String, List<String>> displayNames = results
+				.dynamicallyRegisteredEvents()
+				.map(Event::getTestDescriptor)
+				.collect(Collectors
+						.groupingBy(JsonFileSourceArgumentsProviderTests::testSourceMethodName,
+							Collectors.mapping(TestDescriptor::getDisplayName, Collectors.toList())));
+
+		assertThat(displayNames)
+				.containsOnlyKeys("deconstructCustomerFromArray", "deconstructCustomerMultipleFiles",
+					"deconstructCustomerMultipleFilesComplexType", "singleCustomer", "singleCustomerName",
+					"customDataLocation");
+
+		assertThat(displayNames.get("deconstructCustomerFromArray")).containsExactly("[1] Luke, 172", "[2] Yoda, 66");
+
+		assertThat(displayNames.get("deconstructCustomerMultipleFiles"))
+				.containsExactly("[1] 66, Yoda", "[2] 172, Luke");
+
+		assertThat(displayNames.get("deconstructCustomerMultipleFilesComplexType"))
+				.containsExactly("[1] Yoda, Location{name='unknown'}", "[2] Luke, Location{name='Tatooine'}");
+
+		assertThat(displayNames.get("singleCustomer"))
+				.containsExactly("[1] Customer{name='Luke', height=172}", "[2] Customer{name='Yoda', height=66}");
+
+		assertThat(displayNames.get("singleCustomerName")).containsExactly("[1] Luke", "[2] Yoda");
+
+		assertThat(displayNames.get("customDataLocation"))
+				.containsExactly("[1] Snowspeeder, 4.5", "[2] Imperial Speeder Bike, 3");
```

Comment:
Good job with these tests! Since they already check all the correct values, don't you think we can remove the assertions in the actual tests below? Otherwise, all changes need to be made in two places.

Reply:
We can remove them yes, but then if you want to debug you'll need to write something to test things out.

If new things are added I expect new json files to be used.

Reply:
Got it. It's unfortunate, but makes sense.